nmtwizard/preprocess/operators/similarity_filter.py

```diff
@@ -31,6 +31,6 @@ def _filter(tu):
                 p = random.random()
                 if mode == "soft_sigmoid":
                     norm_v = 1 / (1 + math.exp(-norm_v))
-            return p > norm_v
+            return (p > norm_v, f"Similarity score {norm_v} lower than {p}")
```

Comment:
The message will be formatted for every TU, even when not in verbose mode. Did you consider other ways to only format the message when we actually want to log it?

Reply:
I changed it to only send the message when filter is True and verbose is on. Looks good to you ? The code gets longer, but it's probably ok ? If it's ok I'll change length filters the same way.

Reply:
Indeed, it's the code that is verbose now :). Let's rollback to what you had initially:

```python
return (p > norm_v, f"Similarity score {norm_v} lower than {p}")
```

We'll check the execution profile once this change is released. I don't think it will cause any performance issue.

Reply:
Ok, one last try :) how about this :

`return (p > norm_v, f"Similarity score {norm_v} lower than {p}") if self._verbose else p > norm_v`

This way it won't hurt production performance where verbose shouldn't be set to true.

Reply:
Yep, that's OK. You can also store `p > norm_v` in a variable first, so that the condition is in one place.

Reply:
Changed it for both similarity and length filter, let me know if it looks for you.